src/Players/Player.cpp

```diff
@@ -13,19 +13,35 @@ void Player::onStart(BWAPI::Race race) {
 }
 
 void Player::onFrame() {
-    for (auto& [key, value] : this->armyUnits) {
-        value->onFrame();
-        value->displayInfo();
+	std::chrono::high_resolution_clock::time_point t1 = std::chrono::high_resolution_clock::now();
+	for (auto& [key, value] : this->armyUnits) {
+		value->onFrame();
+	}
+	std::chrono::high_resolution_clock::time_point t2 = std::chrono::high_resolution_clock::now();
+	auto duration = std::chrono::duration_cast<std::chrono::microseconds>(t2 - t1).count();
+	BWAPI::Broodwar->drawTextScreen(0, 60, "Army time: %d ms", duration);
```

Comment:
this can be added inside the if statement, since if there are no units in the map, this should be 0 as well
both can be on the same line, if possible, maybe try to align the time values into columns?